File: app/src/main/java/com/example/meyss/javaretrofitdagger/data/Attack.java
package com.example.meyss.javaretrofitdagger.data;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity()
public class Attack {
    @PrimaryKey
    public int attackId ;
    public String name;
    public String text;
    public  String damage;
    public String pokId;
    public int convertedEnergyCost;

    public int getAttackId() {
        return attackId;
    }

    public String getPokId() {
        return pokId;
    }

    public void setPokId(String pokId) {
        this.pokId = pokId;
    }

    public void setAttackId(int attackId) {
        this.attackId = attackId;
    }

    @Override
    public String toString() {
        return "Attack{" +
                "attackId=" + attackId +
                ", name='" + name + '\'' +
                ", text='" + text + '\'' +
                ", damage='" + damage + '\'' +
                ", pokId='" + pokId + '\'' +
                ", convertedEnergyCost=" + convertedEnergyCost +
                '}';
    }
}

File: app/src/main/java/com/example/meyss/javaretrofitdagger/data/Attack.kt
package com.example.meyss.javaretrofitdagger.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity
class Attack : Serializable{
    @PrimaryKey
    var attackId: Int = 0
    var name: String? = null
    var text: String? = null
    var damage: String? = null
    var pokId: String? = null
    var convertedEnergyCost: Int = 0

    override fun toString(): String {
        return "Attack{" +
                "attackId=" + attackId +
                ", name='" + name + '\''.toString() +
                ", text='" + text + '\''.toString() +
                ", damage='" + damage + '\''.toString() +
                ", pokId='" + pokId + '\''.toString() +
                ", convertedEnergyCost=" + convertedEnergyCost +
                '}'.toString()
    }
}
